### PR TITLE
Fix rpmalloc initializing problem in android

### PR DESCRIPTION
**Pull request by King19931229**
Fix rpmalloc initializing problem in android, compile failure on ios

### CODE REVIEW

**Comment by wolfpld** [comment]
Why is this extra initialization needed? What are the code paths on which you experience problems?

File: client/tracy_rpmalloc.cpp [hunk 1]
```diff
@@ -130,7 +130,9 @@
 #  include <stdio.h>
 #  include <stdlib.h>
 #  if defined(__APPLE__)
-#    include <mach/mach_vm.h>
+#if    !TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+#  include <mach/mach_vm.h>
+#      endif
```
Comment (wolfpld) [inline client/tracy_rpmalloc.cpp:135]: Please fix indentation here.

(Upstream fix: https://github.com/mjansson/rpmalloc/issues/146)

**Comment by King19931229** [comment]
> Why is this extra initialization needed? What are the code paths on which you experience problems?

Here is the call stack I have:
tracy::Callstack(int) TracyCallstack.hpp:87
tracy::Profiler::SendCallstack(int) TracyProfiler.hpp:535
tracy::ScopedZone::ScopedZone(tracy::SourceLocationData const*, int, bool) TracyScoped.hpp:52
FPThreadEvent::Wait(unsigned int, bool) GenericPlatformProcess.cpp:371
FEvent::Wait() Event.h:74
FAsyncTask<RenderAssetInstanceTask::FDoWorkTask>::SyncCompletion() AsyncWork.h:339
FAsyncTask<RenderAssetInstanceTask::FDoWorkTask>::EnsureCompletion(bool) AsyncWork.h:442
FRenderAssetStreamingManager::UpdateResourceStreaming(float, bool) StreamingManagerTexture.cpp:1495
FStreamingManagerCollection::UpdateResourceStreaming(float, bool) ContentStreaming.cpp:866
IStreamingManager::Tick(float, bool) ContentStreaming.cpp:718
FStreamingManagerCollection::Tick(float, bool) ContentStreaming.cpp:836
UGameEngine::Tick(float, bool) GameEngine.cpp:1918
FEngineLoop::Tick() LaunchEngineLoop.cpp:4830

When the flow reach the Callstack function,  it calls the tracy_malloc function which also call rpmalloc function. However it is possible here the rpmalloc is not initialized for this thread. So I add the InitRPMallocThread for avoiding crash. The same reason is for the modification of other positions.

`static tracy_force_inline void* Callstack( int depth )
{
    InitRPMallocThread();

    assert( depth >= 1 && depth < 63 );

    auto trace = (uintptr_t*)tracy_malloc( ( 1 + depth ) * sizeof( uintptr_t ) );
    BacktraceState state = { (void**)(trace+1), (void**)(trace+1+depth) };
    _Unwind_Backtrace( tracy_unwind_callback, &state );

    *trace = (uintptr_t*)state.current - trace + 1;

    return trace;
}`

**Comment by wolfpld** [comment]
This has been solved in another way (b0fc0d5dcc921 and thereabouts).

**Pull request closed without merging**